parsers/src/markup/lib.rs

```diff
@@ -66,36 +67,39 @@ impl<E> Library<E>
                                   templates: &HashMap<String, Template>,
                                   node: &mut Node)
     {
-        for child in node.children.iter_mut() {
-
-            let mut is_empty = None;
-            let test = match child.node_type {
-                NodeType::Template(TemplateData { ref path }) => {
-                    is_empty = Some(path.clone());
-                    templates.get(path)
-                }
-                _ => None
-            };
-
-            match test {
-                Some(found) => {
-                    mem::swap(
-                        child,
-                        &mut Node::from_template(
-                            found,
-                            NodeType::Group
-                        )
-                    );
+        let test = match node.node_type {
+            NodeType::Template(TemplateData { ref path }) => {
+                match templates.get(path) {
+                    None => {
+                        err.log(format!(
+                                "Warning `{}` template name not found", path));
+                        None
+                    }
+                    Some(found) => Some(Node::from_template(found, NodeType::Group)),
                 }
-                // TODO: Warn if the template name is not valid
-                //       (not a data-bindings)
-                None => match is_empty {
-                    Some(name) => err.log(
-                        format!("Warning `{}` template name not found", name)
-                    ),
-                    None => ()
+            }
+            NodeType::UnlinkedRepeat(UnlinkedRepeatData{ref mut iter, ref template_name}) => {
+                match templates.get(template_name) {
+                    None => {
+                        err.log(format!(
+                                "Warning `{}` template name not found for repeat node",
+                                template_name));
+                        None
+                    }
+                    Some(found) => {
+                        let mut child = Node::from_template(found, NodeType::Group);
+                        replace_iterator_name(iter, &mut child);
```

Comment:
This code is wrong, the way the tree is changed is incorrect.
This function should not change the tree that way, the Repeat tag should be handled at the view level.